libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -468,6 +393,91 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 	_variable.annotation().type = type;
 }
 
+void ReferencesResolver::operator()(yul::FunctionDefinition const& _function)
+{
+	m_yulInsideFunction++;
```

Comment:
While this works, I think I'd prefer
```
bool wasInsideFunction = m_yulInsideFunction;
m_yulInsideFunction = true;
this->operator()(_function.body);
m_yulInsideFunction = wasInsideFunction;
```
...

Reply:
that is indeed nicer.. simpler.

Reply:
pushed